File: Practice02/Exercise01/SavingBankManager.cs
namespace Exercise01;

public class SavingBankManager
{
    private List<SavingBank> _accounts = [];

    public void AddAccount(SavingBank newAccount)
    {
        _accounts.Add(newAccount);
    }

    public SavingBank? SearchByCuil(string cuil)
    {
        var nextAccount = _accounts[0];
        var i = 1;
        while (i < _accounts.Count && _accounts[i].Cuil != cuil)
        {
            nextAccount = _accounts[i++];
        }

        return nextAccount.Cuil == cuil ? nextAccount : null;
    }
}

File: practice-02/exercise-01/src/SavingBankManager.ts
import { SavingBank } from './SavingBank'

export class SavingBankManager {
    private accounts: SavingBank[]

    constructor() {
        this.accounts = []
    }

    addAccount(
        accountNumber: string,
        cuil: string,
        lastName: string,
        firstName: string,
        balance: number
    ): void {
        const newAccount = new SavingBank(
            accountNumber,
            cuil,
            lastName,
            firstName,
            balance
        )
        this.accounts.push(newAccount)
    }
    searchAccountByCuil(cuil: string): SavingBank | null {
        let i = 0
        let nextAccount = this.accounts[i]

        while (i < this.accounts.length && nextAccount.getCuil != cuil) {
            nextAccount = this.accounts[++i]
        }

        return nextAccount.getCuil == cuil ? nextAccount : null
    }
}
